Major package Update

theimo1221 opened the pull request:
Following some npm audits, it might be a good idea to update some of stuns packages.

I made sure all my changes resulted in valid unit test completion.

theimo1221 committed:
Upgrade packages

theimo1221 committed:
Lint all

theimo1221 committed:
Remove outdated jest options

theimo1221 commented:
@olleolleolle Could you take a look over this?

olleolleolle commented on test/attributes/stun-address-attribute.js:
```diff
@@ -10,29 +10,15 @@ test('encode', () => {
-    0xa8,
-    1,
-    2 /* IP */,
+    0 /* Reserved */, 0x1 /* Family */, 0xf8, 0x24 /* Port */, 0xc0, 0xa8, 1, 2 /* IP */,
```
This is a linter removing the line comments, making it difficult for a reader to scan.

Can you find ways to build up the array (perhaps in more steps, perf does not matter in this test, but legibility will matter) that the linter will accept, and that lets the code keep the by-line comments?

olleolleolle commented on test/attributes/stun-address-attribute.js:
```diff
@@ -10,29 +10,15 @@ test('encode', () => {
-    0xa8,
-    1,
-    2 /* IP */,
+    0 /* Reserved */, 0x1 /* Family */, 0xf8, 0x24 /* Port */, 0xc0, 0xa8, 1, 2 /* IP */,
```
(Same here)

olleolleolle commented on test/message/decode.js:
```diff
@@ -6,39 +6,11 @@ const { messageType } = require('lib/constants');
-    0x3a,
-    0x9d,
-    0xf7 /* Transaction */,
+    0, 0x01 /* Type */, 0, 12 /* Length */, 0x21, 0x12, 0xa4, 0x42 /* Cookie */, 0xd0, 0x05, 0x58,
```
(Same here in this file)

olleolleolle commented on test/message/encode.js:
```diff
@@ -14,39 +14,11 @@ test('should encode request', () => {
-    0x3a,
-    0x9d,
-    0xf7 /* Transaction */,
+    0, 0x01 /* Type */, 0, 12 /* Length */, 0x21, 0x12, 0xa4, 0x42 /* Cookie */, 0xd0, 0x05, 0x58,
```
(Same here in this file)

olleolleolle reviewed (changes requested):
Super-change! Thanks for taking the time to update the dependencies, that means a lot.

The linter did a good job of making uniform all the files.

The tests have carefully-composed arrays used as Buffer.from arguments, and it would be cool to build them in such a way that the line comments can be kept.

Perhaps like this, perhaps some other way, that lets the text remain on the same line as the value that it talks about.

```js
let input = []
input.push(0x00) // A null there.
```

theimo1221 committed:
Fix issue around upgrading babel

As babel now uses istanbul internally as well, we'd run in an issue without an specific name here

theimo1221 committed:
update package-lock.json (including audit fix)

theimo1221 committed:
Change array declaration to have proper inline comments

theimo1221 commented:
@olleolleolle Thanks for the quick review! Following your suggestion I updated the PR.

olleolleolle commented:
(looks good to me!)

theimo1221 commented:
> (looks good to me!)

I thought you might have the necessary permissions to menge this PR and run a release...

Do you know how to reach one of the owners?

theimo1221 commented:
@reklatsmasters Could you look into this please

reklatsmasters commented:
A very controversial change, only brackets and bytes. Rejected.

theimo1221 commented:
Hello @reklatsmasters,

I'm sorry but this PR doesn't just include brackets and bytes.
The main purpose are those package updates and afterwards the correctness in regards to the defined linting rules.

@olleolleolle am I missing something here?
